protocol.go

```diff
@@ -155,3 +157,39 @@ func (r *errorTranslatingReceiver) Receive(msg any) error {
 func (r *errorTranslatingReceiver) Close() error {
 	return r.fromWire(r.Receiver.Close())
 }
+
+func sortedAcceptPostValue(handlers []protocolHandler) string {
+	var max int
+	for _, handler := range handlers {
+		max += len(handler.ContentTypes())
+	}
+	seen := make(map[string]struct{}, max)
```

Comment:
why not just add all content types to `seen`, then convert to a slice, and sort the slice?

Reply:
It's about the same amount of code, since there's no convenient set type with an `Elements() [T]` method, and it makes more passes over the data. Don't mind changing if you'd prefer it that way.